include/rocksdb/file_system.h

```diff
@@ -733,6 +750,19 @@ struct FSReadRequest {
   IOStatus status;
 };
 
+// A read IO response structure for use by asynchronously Read APIs.
+struct FSReadResponse {
+  // File offset in bytes
+  uint64_t offset;
+
+  // Output parameter set by Read APIs to point to the data buffer, and
+  // the number of valid bytes.
+  Slice result;
```

Comment:
As @mrambacher said, it would be good to clarify the lifecycle of ```scratch``` in ```FSReadRequest```, and ```result```. The ```scratch``` buffer is allocated by RocksDB and would exist until the IO is completed and callback has been called. I think we should also clarify that the ```result``` Slice should point to ```scratch```, i.e the data should always be read into ```scratch```. I don't see any case where the FS implementation would allocate its own buffer and pass the pointer in ```result```.

Reply:
Added the details in `FSReadRequest` and `FSReadResponse`. LMK if anything needs to be updated or any detail is missing.